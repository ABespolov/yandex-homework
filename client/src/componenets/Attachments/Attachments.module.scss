@import "../../mixins";

.attachments {
  display: flex;
  flex-flow: column;
  padding: 0 20px;
  font-weight: 600;
  background: #e6e6e6;

  .linkAttachment {
    display: flex;
    justify-content: start;
    align-items: center;
    margin-right: -5px;
    margin-left: -5px;
    padding: 14px 10px;

    @include smoothResize(font-size, 16, 18);
  }

  .linkAttachment:not(:last-child) {
    border-bottom: 1px solid #fff;
  }

  .images {
    display: flex;
    justify-content: space-between;
    width: 100%;

    .imageAttachment {
      display: flex;
      align-items: center;
      padding: 12px 0;
      font-size: 0;

      .attachment img {
        width: 64px;
        height: 64px;
        margin-left: 24px;
        border-radius: 4px;
      }
    }
  }

  .linkIcon {
    margin-right: 12px;
  }
}
