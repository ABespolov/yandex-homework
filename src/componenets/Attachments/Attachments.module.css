.attachments {
  display: flex;
  flex-flow: column;
  padding: 0 20px;
  font-weight: 600;
  background: #e6e6e6;
}

.attachment,
.linkAttachment {
  display: flex;
  justify-content: end;
  align-items: center;
  height: 58px;
  margin-right: -5px;
  margin-left: -5px;
  padding: 0 10px;
}

.images {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.attachments .linkAttachment:not(:last-child) {
  border-bottom: 1px solid #fff;
}

.imageAttachment {
  display: flex;
  align-items: center;
  padding: 25px 0;
}

.imageAttachment .attachment img {
  width: 64px;
  height: 64px;
  margin-left: 20px;
}

.linkIcon {
  margin-right: 12px;
}
