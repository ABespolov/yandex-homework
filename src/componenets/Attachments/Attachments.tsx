import React from "react";
import linkIcon from "../../icons/link.svg";
import imageIcon from "../../icons/picture.svg";
import styles from "./Attachments.module.scss";

interface Attachments {
  attachments: Array<{
    type: string;
    url: string;
  }>;
}

export const Attachments: React.FC<Attachments> = ({attachments}) => {
  const links = attachments && attachments.filter((item) => item.type === "link");
  const images = attachments && attachments.filter((item) => item.type === "image");
  const lnk = links && links.map((item) => (
    <div className={styles.linkAttachment}>
      <img className={styles.linkIcon} src={linkIcon} alt=""/>
      <span>{item.url.replace(/(^.+\/\/)|(\/.+)/g, "")}</span>
    </div>
  ));
  const img = images && images.map((item) => (
    <div className={styles.attachment}>
      <img src={item.url} alt=""/>
    </div>
  ));
  return (
    <div className={styles.attachments}>
      {img.length > 0 ?
        <div className={styles.images}>
          <img src={imageIcon} alt=""/>
          <div className={styles.imageAttachment}>
            {img}
          </div>
        </div>
        : null}
      {lnk}
    </div>
  );
};
