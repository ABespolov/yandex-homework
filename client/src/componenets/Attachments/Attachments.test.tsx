import {configure, shallow} from "enzyme";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import {Attachments} from "./Attachments";

configure({ adapter: new Adapter() });

describe("Attachments:", () => {
    it("Attachments render: ", () => {
        const items = [{type: 'link', url: 'link1'}];
        const component = shallow(<Attachments attachments={items}/>);
        expect(component).toMatchSnapshot();
    });
});
