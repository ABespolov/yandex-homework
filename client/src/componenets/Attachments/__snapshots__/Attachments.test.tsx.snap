// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachments: Attachments render:  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Attachments
    attachments={
      Array [
        Object {
          "type": "link",
          "url": "link1",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        Array [
          <div
            className="linkAttachment"
          >
            <img
              alt=""
              className="linkIcon"
              src="link.svg"
            />
            <span>
              link1
            </span>
          </div>,
        ],
      ],
      "className": "attachments",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              className="linkIcon"
              src="link.svg"
            />,
            <span>
              link1
            </span>,
          ],
          "className": "linkAttachment",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "linkIcon",
              "src": "link.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "link1",
            },
            "ref": null,
            "rendered": "link1",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          Array [
            <div
              className="linkAttachment"
            >
              <img
                alt=""
                className="linkIcon"
                src="link.svg"
              />
              <span>
                link1
              </span>
            </div>,
          ],
        ],
        "className": "attachments",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                className="linkIcon"
                src="link.svg"
              />,
              <span>
                link1
              </span>,
            ],
            "className": "linkAttachment",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "linkIcon",
                "src": "link.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "link1",
              },
              "ref": null,
              "rendered": "link1",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
